http/typescript/fetch/src/middlewares/middlewareFactory.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module MiddlewareFactory
+ */
+
+import { FetchRequestInfo, FetchRequestInit, FetchResponse } from "../utils/fetchDefinitions";
+import { CustomFetchHandler } from "./customFetchHandler";
+import { DefaultFetchHandler } from "./defaultFetchHandler";
+import { Middleware } from "./middleware";
+import { RedirectHandlerOptions } from "./options/redirectHandlerOption";
+import { RetryHandlerOptions } from "./options/retryHandlerOptions";
+import { RedirectHandler } from "./redirectHandler";
+import { RetryHandler } from "./retryHandler";
+
+/**
+ * @class
+ * Class containing function(s) related to the middleware pipelines.
+ */
+export class MiddlewareFactory {
```

Comment:
The middleware chain for browser and node is different. Browsers redirect on their own and do not take in the redirect handler. 

Maintaining this separate files for browser and node. It is best to separate the minimum code into separate node and browser files so that  we do not duplicate a lot of code.

Reply:
interesting, I didn't remember that subtle difference. Ultimately how do you see the initialization happen then? In the service lib client/graph core have a test to see whether the code is running in browser or in node and call the appropriate method?

Reply:
The browser field in package.json contains the map of browser replacements "node file :browser file". 

For a browser, if the code is something `const client = new HttpClient();` which calls the `MiddlewareFactory`. The bundler takes care of the browser field in package.json and calls the browser specific `MiddlewareFactory`. 

The browser field in package.json is the standard that is used by bundlers like rollup, webpack. 

I have set up the browser tests so that we rollup the code and tests with browser config set as true and test  if the browser specific file/code is loaded.